src/validation.cpp

```diff
@@ -4824,6 +4824,28 @@ std::vector<Chainstate*> ChainstateManager::GetAll()
     return out;
 }
 
+std::vector<Chainstate*> ChainstateManager::GetAllForBlockDownload()
+{
+    AssertLockHeld(::cs_main);
+    std::vector<Chainstate*> out;
+
+    bool snapshot_in_ibd{
+        m_snapshot_chainstate && m_snapshot_chainstate->IsInitialBlockDownload()};
+
+    if (m_snapshot_chainstate) {
+        out.push_back(m_snapshot_chainstate.get());
+    }
+    // Exclude the ibd chainstate (which is in the background if we have a snapshot
+    // chainstate active) if the snapshot chainstate is in IBD because the main
+    // priority is getting the active chain to network tip.
+    if (!IsSnapshotValidated() && !snapshot_in_ibd && m_ibd_chainstate) {
+        out.push_back(m_ibd_chainstate.get());
+    }
+
+    assert(out.size() > 0);
```

Comment:
I'm likely misunderstanding something.
Imagine a snapshot doesn't exist. Then, if `m_ibd_chainstate == false`, `out` will be empty, and this would trigger an issue?

So, if the caller must not call it in this case, this requirement probably should be documented?